backend/app/api/crypto.py

```
@@ -704,6 +742,113 @@ async def delete_crypto_transaction(
         raise HTTPException(status_code=500, detail=f"Failed to delete transaction: {str(e)}")
 
 
+# Wallet Sync Status Endpoint
+
+@router.get("/portfolios/{portfolio_id}/wallet-sync-status")
+async def get_wallet_sync_status(
+    portfolio_id: int,
+    db: AsyncSession = Depends(get_db)
```

Ruff (0.14.0)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)